services/app-web/src/components/FileUpload/FileUpload.tsx

```diff
@@ -94,8 +98,17 @@ export const FileUpload = ({
     const inputRequired = inputProps['aria-required'] || inputProps.required
     // update fileItems in parent
     React.useEffect(() => {
-        onFileItemsUpdate({ fileItems })
-    }, [fileItems, onFileItemsUpdate])
+        if (JSON.stringify(fileItems) !== JSON.stringify(previousFileItems)) {
+            onFileItemsUpdate({ fileItems })
+        }
+    }, [fileItems, previousFileItems, onFileItemsUpdate])
```

Comment:
@haworku Here is a change in FIleUpload that will need some extra scrutiny. Because `onFileItemsUpdate` is actually setting the state in `RateDetials` component, it was causing an infinite re-render when there was multiple rate certifications. Checking the `stringify` data of `fileItems` and `previousItems` prevented this use effect from calling `onFileItemsUpdate` unless there was an actual change in files or file data.